packages/@aws-cdk/aws-iam/lib/policy.ts

```diff
@@ -260,3 +265,30 @@ export class Policy extends Resource implements IPolicy {
     return this.groups.length + this.users.length + this.roles.length > 0;
   }
 }
+
+/**
+ * A `grantPrincipal` of Policy.
+ */
+class PolicyGrantPrincipal implements IPrincipal {
+  public readonly assumeRoleAction = 'sts:AssumeRole';
+  public readonly grantPrincipal: IPrincipal;
+  public readonly principalAccount?: string;
+
+  constructor(private _policy: Policy) {
+    this.grantPrincipal = this;
+    this.principalAccount = _policy.env.account;
+  }
+
+  public get policyFragment(): PrincipalPolicyFragment {
+    throw new Error(`Cannot use a Policy ${this._policy.node.path} as the 'Principal' or 'NotPrincipal' in an IAM Policy`);
```

Comment:
```suggestion
    throw new Error(`Cannot use a Policy '${this._policy.node.path}' as the 'Principal' or 'NotPrincipal' in an IAM Policy`);
```

I still don't understand why this is the case